src/extension.ts

```diff
@@ -56,22 +57,32 @@ export async function activate(context: ExtensionContext) {
 		if (event === undefined) {
 			return;
 		}
+		if (event.document.languageId !== 'inmanta' || (event.document.uri.scheme !== 'file')) {
+			return;
+		}
 		const uri = event.document.uri;
 		let folder = workspace.getWorkspaceFolder(uri);
 		folder = getOuterMostWorkspaceFolder(folder);
-
+		// Update the button visibility when the active editor changes
+		pythonExtensionInstance.updateInmantaEnvVisibility(folder.uri);
 		if (folder === lastActiveFolder) {
 			return;
 		}
 		lastActiveFolder = folder;
 		const languageServer = languageServers.get(folder.uri.toString());
 
 		// Update the button visibility when the active editor changes
-		pythonExtensionInstance.updateInmantaEnvVisibility(folder);
+		pythonExtensionInstance.updateInmantaEnvVisibility(folder.uri);
 		inmantaCommands.registerCommands(languageServer);
 	}
 
 	async function didOpenTextDocument(document: TextDocument): Promise<void> {
+		// We are only interested in .cf files
+		if (document.languageId !== 'inmanta' || (document.uri.scheme !== 'file')) {
+			return;
+		}
+		
+		pythonExtensionInstance.updateInmantaEnvVisibility(document.uri)
```

Comment:
I'm not entirely sure how to interpret this. What would happen if you have an inmanta file open and then open a Python file or the `requirements.txt`?

Reply:
I get another venv box with another venv selected when I do this.

Reply:
What's your setting value for "python.interpreter.infoVisibility" ?

Reply:
`onPythonRelated`

Reply:
This is the behavior I have: 

[Screencast from 17-05-2023 10:53:05.webm](https://github.com/inmanta/vscode-inmanta/assets/98876549/e0e6db74-dda7-4a4a-b6f4-77338319938f)

Reply:
Did you install the ls in editable mode ?